uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -216,7 +223,40 @@ define([
        * @return {Boolean} true is user can upload more else false
        */
       this.canUploadMore = function () {
-        return (this.request.files.length + this.request.fileUploader.queue.length) < sharedSettings.fileUploader.queueLimit;
+        var filesWithoutSoftDelete = _.filter(this.request.files, function (file) {
+          return !file.toBeDeleted;
+        });
+
+        return (filesWithoutSoftDelete.length + this.request.fileUploader.queue.length) < sharedSettings.fileUploader.queueLimit;
+      };
+
+      /**
+       * Checks if user can see file list in files tab. It needs to check for
+       * soft delete flag on leave request as well for files.
+       *
+       * @return {Boolean} true is user can see file list
+       */
+      this.canShowFileList = function () {
```

Comment:
Right so it seems you need `canUploadMore()`, but there really seem to be some duplicate logic here. So i think it would be beneficial to add the `getFilesCount` method and use *that* in `canUploadMore()`, something like
```js
this.canUploadMore = function () {
  return getFilesCount() < sharedSettings.fileUploader.queueLimit;
}
```
(i don't think `getFilesCount` need to be a public method

for "canShowFileList", do you really need it? in the ui, can't you just do something like `ng-if="getFilesCount() > 0"`